Implemented basic grid for the spreadsheet

Krihag opened the pull request:
Just basic grid made with JS, but should be flexible in using it (since we dont know design yet)

itsmaik committed:
Delete .env

itsmaik committed:
Updates  README.md file

Anne-Serine committed:
basic html structure setup

Krihag committed:
Implemented basic grid for the spreadsheet

Krihag committed:
small fix for adding ID for each input

FP22FD commented on src/js/spreadsheet/cell.js:
```diff
@@ -0,0 +1,14 @@
+export default function cell(col, row) {
+  const cellContainer = createEle('div', 'relative');
+  const cellInput = createEle(
+    'input',
```
Hi, I was thinking if it's a good idea to have all cells as a input, it looks a pretty expensive DOM. My suggestion is to change to a DIV and I will add the input on click/focus.
But I also think that using a TABLE is more semantically correct and easier to work with. Hence each td is a cell. a tr a row, and you get for free all the role attributes.

Krihag replied:
Good idea. Feel free to implement thoose changes

FP22FD replied:
Don't worry, it was just a suggestion to make a more professional application, but it's not our goal now. Take your time, we have time until Friday 👍I can work around it.

Krihag replied:
It was a good suggestion and I will add it 👍

Krihag replied:
I added the changes. How is it now?

Krihag committed:
Changes to make code abit more semantic (suggestion from Fernanda)

Also removed input for each row (currently a TD element only)

Krihag committed:
added dataset values for col and row to be easily sellected

Krihag committed:
Update cell.js

Krihag committed:
added height and width as arguements

gets default values based on user screen if not specified.

Anne-Serine committed:
Adjust html and mount spreadsheet on an id instead of body

Anne-Serine committed:
Merge branch 'workflow' into spreadsheet-grid

Anne-Serine commented:
I took the liberty to adjust the html a bit and make the spreadsheet mount on an id instead of the body. 
Just run `git pull` to get the latest updates locally.

Krihag committed:
started some error handling to check for valid input numbers + added some descriptions

Also added so that the spreadsheet will be returned instead of appended to a specific element to make more reusable.

Krihag committed:
added usecase if needed more than 26 columns (up to 676) + js docs

itsmaik commented on src/js/main.js:
```diff
@@ -0,0 +1,8 @@
+import spreadsheet from './spreadsheet';
+import userColsAndRows from './helpers/userColsAndRows';
+
+const spreadsheetContainer = document.querySelector('#spreadsheetContainer');
+
+const [cols, rows] = userColsAndRows();
+
+spreadsheetContainer.append(spreadsheet(cols, rows));
```
main.js should be only used for importing files, all the other function and variables should be in its own file.

itsmaik replied:
main.js should be only used for importing files, all the other function and variables should be in its own file.

itsmaik commented on src/js/helpers/ifValidNumber.js:
```diff
@@ -0,0 +1,30 @@
+/**
+ * Checks if the given values for columns and rows are valid numbers.
+ * If the values are not valid, it throws an error.
```
```
 * Checks if the given values for columns and rows are valid numbers.
 * If the values are not valid, it throws an error.
This is a test review 
```
